phonetest.py

```diff
@@ -200,18 +201,93 @@ def lookup(cls, phoneid, config_file, chunk):
     @classmethod
     def match(cls, tests=None, test_name=None, phoneid=None,
               config_file=None, job_guid=None,
-              build_url=None):
+              build_url=None, branch=None, revision=None):
 
         logger.debug('PhoneTest.match(tests: %s, test_name: %s, phoneid: %s, '
                      'config_file: %s, job_guid: %s, '
-                     'build_url: %s' % (tests, test_name, phoneid,
-                                        config_file, job_guid,
-                                        build_url))
+                     'build_url: %s, branch: %s, revision: %s' % (
+                      tests, test_name, phoneid, config_file, job_guid,
+                      build_url, branch, revision))
+
         matches = []
         if not tests:
             tests = [PhoneTest.instances[key] for key in PhoneTest.instances.keys()]
 
+        # See if we need to check changeset files so we don't hit the hg
+        # server unnecessarily.
+        has_run_if_changed = False
         for test in tests:
+            if test.run_if_changed:
+                has_run_if_changed = True
+                break
```

Comment:
I'd rather do this outside of PhoneTest.match and to pass the changed files to PhoneTest.match instead of passing a revision and looking the changed files up inside of match.